test/integration/PoolSwap.t.sol

```diff
@@ -20,65 +20,82 @@ abstract contract PoolSwapIntegrationTest is Test, GasSnapshot {
   Vm.Wallet swapper = vm.createWallet('swapper');
   Vm.Wallet swapperInverse = vm.createWallet('swapperInverse');
 
-  uint256 public constant HUNDRED_UNITS = 100 ether;
+  /**
+   * For the simplicity of this test, a 1000 DAI:1 ETH reference quote is used.
+   * A weight distribution of 80% DAI and 20% ETH is used.
+   * To achieve the reference quote, the pool should have 4000 DAI and 1 ETH.
+   *
+   * On the one swap, 100 DAI is swapped for ~0.1 ETH.
+   * On the inverse swap, 0.1 ETH is swapped for ~100 DAI.
+   */
+
+  // unit amounts
+  uint256 public constant ONE_TENTH_UNIT = 0.1 ether;
   uint256 public constant ONE_UNIT = 1 ether;
+  uint256 public constant HUNDRED_UNITS = 100 ether;
+  uint256 public constant FOUR_THOUSAND_UNITS = 4000 ether;
+
+  // pool amounts
+  uint256 public constant DAI_LP_AMOUNT = FOUR_THOUSAND_UNITS;
+  uint256 public constant WETH_LP_AMOUNT = ONE_UNIT;
 
-  // NOTE: hardcoded from test result
-  uint256 public constant WETH_AMOUNT = 0.096397921069149814e18;
-  uint256 public constant DAI_AMOUNT = 0.5e18;
+  // swap amounts IN
+  uint256 public constant DAI_AMOUNT = HUNDRED_UNITS;
+  uint256 public constant WETH_AMOUNT_INVERSE = ONE_TENTH_UNIT;
 
-  uint256 public constant WETH_AMOUNT_INVERSE = 0.1e18;
-  // NOTE: hardcoded from test result
-  uint256 public constant DAI_AMOUNT_INVERSE = 0.316986296266343639e18;
+  // swap amounts OUT
+  // NOTE: amounts OUT are hardcoded from test result
+  uint256 public constant WETH_OUT_AMOUNT = 94_049_266_814_811_022; // 0.094 ETH
+  uint256 public constant DAI_OUT_AMOUNT_INVERSE = 94_183_552_501_642_552_000; // 94.1 DAI
 
   function setUp() public virtual {
     vm.createSelectFork('mainnet', 20_012_063);
 
     factory = _deployFactory();
 
-    deal(address(dai), lp, HUNDRED_UNITS);
-    deal(address(weth), lp, HUNDRED_UNITS);
+    deal(address(dai), lp, DAI_LP_AMOUNT);
+    deal(address(weth), lp, WETH_LP_AMOUNT);
 
-    deal(address(dai), swapper.addr, ONE_UNIT);
-    deal(address(weth), swapperInverse.addr, ONE_UNIT);
+    deal(address(dai), swapper.addr, DAI_AMOUNT);
+    deal(address(weth), swapperInverse.addr, WETH_AMOUNT_INVERSE);
 
     vm.startPrank(lp);
     pool = factory.newBPool();
 
     dai.approve(address(pool), type(uint256).max);
     weth.approve(address(pool), type(uint256).max);
-    pool.bind(address(dai), ONE_UNIT, 2e18); // 20% weight
-    pool.bind(address(weth), ONE_UNIT, 8e18); // 80% weight
+    pool.bind(address(dai), DAI_LP_AMOUNT, 8e18); // 80% weight
```

Comment:
when reading this the first time I had a hard time figuring out when was the liquidity added, since there are no explicit join calls. This makes it easier to follow :sparkles: